op-challenger/challenger/output_test.go

```diff
@@ -0,0 +1,172 @@
+package challenger
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/metrics"
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/testlog"
+)
+
+func TestChallenger_ValidateOutput_PanicsWithoutRollupClient(t *testing.T) {
+	challenger := Challenger{}
+	require.Panics(t, func() {
+		_, _, err := challenger.ValidateOutput(context.Background(), nil, eth.Bytes32{})
+		require.NoError(t, err)
+	})
+}
+
+func TestChallenger_ValidateOutput_PanicsWithoutLog(t *testing.T) {
+	outputApi := &mockOutputApi{}
+	challenger := Challenger{
+		rollupClient: outputApi,
+	}
+	require.Panics(t, func() {
+		_, _, err := challenger.ValidateOutput(context.Background(), nil, eth.Bytes32{})
+		require.NoError(t, err)
+	})
+}
+
+func TestChallenger_ValidateOutput_PanicsWithoutMetricer(t *testing.T) {
+	outputApi := mockOutputApi{}
+	log := testlog.Logger(t, log.LvlError)
+	metr := metrics.NewMetrics("test")
+	challenger := &Challenger{
+		rollupClient:   &outputApi,
+		log:            log,
+		metr:           metr,
+		networkTimeout: time.Duration(5) * time.Second,
+	}
+	require.Panics(t, func() {
+		_, _, err := challenger.ValidateOutput(context.Background(), nil, eth.Bytes32{})
+		require.NoError(t, err)
+	})
+}
+
+func TestChallenger_ValidateOutput_RollupClientErrors(t *testing.T) {
+	output := eth.OutputResponse{
+		Version:    supportedL2OutputVersion,
+		OutputRoot: eth.Bytes32{},
+		BlockRef:   eth.L2BlockRef{},
+	}
+
+	outputApi := newMockOutputApi(output, true)
+	log := testlog.Logger(t, log.LvlError)
+	metr := metrics.NewMetrics("test")
+	challenger := Challenger{
+		rollupClient:   outputApi,
+		log:            log,
+		metr:           metr,
+		networkTimeout: time.Duration(5) * time.Second,
+	}
```

Comment:
Could probably extract a setup function to avoid duplication - something like:
```
func (output eth.OutputResponse, errors bool) (*Challenger, *mockOutputApi)
```

Probably worth introducing a no-op metrics implementation like we have in op-node as well rather than having to create the full metrics implementation for every test.